crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -55,11 +31,6 @@ pub(crate) enum Instruction {
     // Binary Operations
     Binary(Binary),
 
-    // Unary Operations
```

Comment:
Why is the Cast instruction removed?

Reply:
I think it is duplicated by the Truncate instruction and we should only have one or the other.

I can move it into a separate PR since it is unrelated to the map changes.

Reply:
Oh I see, if we can only cast to integers and the truncate is doing a casting implicitly then that makes sense. A separate PR would be good please moreso since its a change from the existing IR